### PR TITLE
Chore: Refactor_wallet_manager

### PR DESCRIPTION
**Pull request by k9ert**
This is a preparation of making the `WalletManager ` aware of the active node via a session-attribute.

relates to #1661

### CODE REVIEW

**Comment by k9ert** [comment]
The experimental session-feature used in the `spec_multiuser_sessions.js` from cypress doesn't seem to work for us. I've created:
https://github.com/cypress-io/cypress/issues/21138

**Comment by k9ert** [comment]
This is ready for review. I'd like to get this in part by part. I could continue here but i think it's better to merge it as is.

File: cypress/integration/spec_multiuser_sessions.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+describe('Test the actions in UTXO list', () => {
```
Comment (moneymanolis) [inline cypress/integration/spec_multiuser_sessions.js:1]: This is probably from copy&pasting, change this to provide more info on what the test is about.

File: cypress/integration/spec_multiuser_sessions.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    // See https://github.com/cypress-io/cypress/issues/21138
+    if (Cypress.env("CI")) {
+        const login = (name,password) => {
```
Comment (moneymanolis) [inline cypress/integration/spec_multiuser_sessions.js:4]: Perhaps I am overlooking sth. but where is the name variable set?

**Review by moneymanolis — APPROVED** [review]

**Comment by djpnewton** [comment]
So this does not yet make the wallet manager session aware? It is just refactoring to use less instance variables such that when the wallet manager is session aware things like the `wallets_update_list` dict will not contain wallets from multiple different sessions?

**Comment by k9ert** [comment]
> So this does not yet make the wallet manager session aware? 
No. It's also not planned to make it session-aware.

> It is just refactoring to use less instance variables such that when the wallet manager is 
> session aware things like the wallets_update_list dict will not contain wallets from multiple different sessions?

This was just a side_effect of that change. The main point from my point of view is that the `WalletManager` no longer throws away all it's data when the Node gets changed. This is a prerequisite to requesting data from an instance which is session-aware.

File: cypress/integration/spec_multiuser_sessions.js (hunk 2 above)
Comment (k9ert) [reply cypress/integration/spec_multiuser_sessions.js:4]: It's defining a function and the name is the parameter here. It's initialized with the arguments at call time, e.g. in line 39.

**Pull request merged**